src/utils.ts

```
@@ -461,29 +461,42 @@ export function safeParseUrl(url: string, baseUrl?: string): URL {
   }
 }
 
+/**
+ * Returns an https: schema URL, or throws a BusinessError
+ * @param url an absolute or relative URL
+ * @param baseUrl the baseUrl to use if url is relative
+ * @return the URL instance
+ * @throws BusinessError if the URL is invalid
+ */
 export function assertHttpsUrl(
   url: string,
-  baseUrl: string = location.href
+  // Don't default baseUrl to location.href here. API calls are always routed through a chrome-extension:// page (e.g.,
+  // the background page. So they would always be flagged as having an invalid schema)
+  baseUrl?: string
```

What does this mean? If API calls are routed via the background page, then an unset URL base _is_ a `chrome-extension://` page whether we set it or not. This sounds like it just silences errors on relative URLs that don't have a base

---

In practice though this specific change may be correct 👍  because it changes the error from `Unsupported protocol: chrome-extension:` to `Invalid URL: /api/call` (but this error isn't clear either)

---

>  If API calls are routed via the background page, then an unset URL base is a chrome-extension:// page whether we set it or not

All calls from the `/get` and `/http` bricks need to either be 1) absolute, or 2) relative and using a service that defines the base URL.

There's various checks for this at:
- http://github.com/pixiebrix/pixiebrix-extension/blob/fd14c1096222ff7d769b95626bcc44a6f021be5f/src/background/requests.ts#L230-L230
- http://github.com/pixiebrix/pixiebrix-extension/blob/fd14c1096222ff7d769b95626bcc44a6f021be5f/src/services/factory.ts#L243-L243
- http://github.com/pixiebrix/pixiebrix-extension/blob/fd14c1096222ff7d769b95626bcc44a6f021be5f/src/services/factory.ts#L210-L210

You're correct that the unset base URL is `chrome-extension://`. The issue is that the `assertHttpsUrl` method produces a misleading error. The error is not a BusinessError, but an Application Error - in order for a relative URL to have been passed there must be some error in our application code

> Maybe it should just be mandatory so assertHttpsUrl doesn't have to concern itself with this

I agree it probably makes sense to make the argument mandatory with type `string | null`